### PR TITLE
Add basic type annotations for YAML load functions

### PR DESCRIPTION
**Pull request by Wilfred**
These are probably the most common YAML functions, so this should be
useful even without type declarations on the rest of the package.

This is my first contribution to typeshed, so please let me know if this patch makes sense. I tried to find a more specific type for the `stream` argument (I know it supports both strings and file objects) but I couldn't find a definitive list of the types supported.

### CODE REVIEW

File: third_party/2and3/yaml/__init__.pyi [hunk 1]
```diff
@@ -14,10 +14,10 @@ def scan(stream, Loader=...): ...
 def parse(stream, Loader=...): ...
 def compose(stream, Loader=...): ...
 def compose_all(stream, Loader=...): ...
-def load(stream, Loader=...): ...
-def load_all(stream, Loader=...): ...
-def safe_load(stream): ...
-def safe_load_all(stream): ...
+def load(stream, Loader=...): ...  # type: (Any, Loader) -> Any
```
Comment (matthiaskramm) [inline third_party/2and3/yaml/__init__.pyi]: It seems weird to use comment-based annotations here. Why not just

```
def load(stream: Any, Loader: Loader = ...) -> Any: ...
```
?

Comment (JelleZijlstra) [reply third_party/2and3/yaml/__init__.pyi]: Also, for the stream, could something like `Union[str, IO[str]]` work?

Comment (Wilfred) [reply third_party/2and3/yaml/__init__.pyi]: Updated.

**Comment by Wilfred** [comment]
@matthiaskramm do I need to make any further changes to get this merged?

**Comment by matthiaskramm** [comment]
Looks good to me. Merging.

File: third_party/2and3/yaml/__init__.pyi [hunk 2]
```diff
@@ -14,10 +14,10 @@ def scan(stream, Loader=...): ...
 def parse(stream, Loader=...): ...
 def compose(stream, Loader=...): ...
 def compose_all(stream, Loader=...): ...
-def load(stream, Loader=...): ...
-def load_all(stream, Loader=...): ...
-def safe_load(stream): ...
-def safe_load_all(stream): ...
+def load(stream: Union[str, IO[str]], Loader: Loader=...) -> Any: ...
```
Comment (gvanrossum) [inline third_party/2and3/yaml/__init__.pyi]: I discovered a problem with the annotation for `Loader` here. IIUC it's intended that this argument can be an instance of `BaseLoader` or `SafeLoader` as well. But given the inheritance structure in `loader.pyi` that doesn't work -- if I pass an instance of `SafeLoader` I get a type error. Maybe you can fix this by declaring this argument (here and below) as `Union[BaseLoader, SafeLoader, Loader]`? (Do double check that I got this right -- I'm not a user of yaml myself, but I found this caused a regression in an internal Dropbox codebase that uses it.)

Comment (gvanrossum) [reply third_party/2and3/yaml/__init__.pyi]: I fixed this in #1695.

**Pull request merged**